WordPress/Classes/Login/WordPressDotComAuthenticator.swift

```diff
@@ -191,7 +193,42 @@ struct WordPressDotComAuthenticator {
 
         let callbackURL = try await authorize(from: viewController, url: authorizeURL, prefersEphemeralWebBrowserSession: prefersEphemeralWebBrowserSession)
 
-        return try await handleAuthorizeCallbackURL(callbackURL, clientId: clientId, clientSecret: clientSecret, redirectURI: redirectURI)
+        let theError: AuthenticationError
+        do {
+            return try await handleAuthorizeCallbackURL(callbackURL, clientId: clientId, clientSecret: clientSecret, redirectURI: redirectURI)
+        } catch {
+            theError = error
+        }
+
+        guard case .loginDenied = theError, recoverDenyAccess else {
+            throw theError
+        }
+
+        // Try to re-authenticate when user taps the "Deny" button.
+        let reLogin = await withCheckedContinuation { continuation in
+            DispatchQueue.main.async {
+                let format = NSLocalizedString("wpComLogin.loginDenied.message", value: "You can sign in with a different account if you need a different one. Tap \"%@\" to start.", comment: "Message shown when user denies WordPress.com login, offering option to try with different account")
```

Comment:
Please extract the localizable strings into `Strings`, and I'd also extracting the "reLogin" code into a separate method to make it easier to follow the overall flow of the `authenticate` method.

Reply:
👍 Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/24215/commits/10dc0cc4874305993934c6af1a163ca47336c020